packages/nns/src/errors/ledger.errors.ts

```diff
@@ -83,8 +84,8 @@ export const mapTransferProtoError = (responseBytes: Error): TransferError => {
         /debit account.*, current balance: (\d*(\.\d*)?)/
       );
       if (m && m.length > 1) {
-        const balance = ICP.fromString(m[1]);
-        if (balance instanceof ICP) {
+        const balance = convertStringToE8s(m[1]);
+        if (typeof balance === "bigint") {
```

Comment:
since we are already there, should we modify `convertStringToE8s` to throw error so that we don't have to validate the the return type?

Reply:
That would also mean even more refactor in the nns-dapp. I think we have enough refactor as is.

But I agree that's something we should do.

I also kept `toProto` in the TokenAmount to reduce the amount of refactor and to make it simpler.

Reply:
Good point, let's do later on